server/src/main.rs

```diff
@@ -0,0 +1,156 @@
+            if let Ok(signature_bytes) = Signature::from_bytes(&signature_vec) {
+                if public_key.verify(&payload.as_bytes(), &signature_bytes).is_ok() {
+                    return true
+                }
+            }
+        }
+    }
+    return false
+}
+
+fn respond_success (is_verified: bool) -> hyper::Response<Body> {
+    // construct response based on verification status
+    match is_verified {
+        true => {
+            Response::builder()
+                .status(StatusCode::OK)
+                .body(Body::empty())
+                .unwrap()
+        },
+        _ => {
+            Response::builder()
+                .status(StatusCode::UNAUTHORIZED)
+                .body(Body::empty())
+                .unwrap()
+        }
+    }
+}
+
+fn read_hp_pubkey() -> PublicKey {
+    let hpos_state_path = env::var("HPOS_STATE_PATH").expect("HPOS_STATE_PATH environmental variable is not present");
+
+    println!("Reading HP Admin Public Key from {}.", hpos_state_path);
+
+    // Read from path
+    let contents = fs::read(hpos_state_path)
+        .expect("Something went wrong reading HP Admin Public Key from file");
+
+    // Parse content
+    let hpos_state: State = serde_json::from_slice(&contents).unwrap();
+    hpos_state.get_admin_public_key().expect("HP Admin Public key seems to be corrupted")
+}
+
+fn main() {
+    // Listen on http socket port 2884 - "auth" in phonespell
+    let listen_address = ([127,0,0,1], 2884).into();
+
+    // Trigger lazy static to see if HP_PUBLIC_KEY assignment creates panic
+    let _ = &*HP_PUBLIC_KEY;
+
+    // Create a `Service` from servicing function
+    let new_svc = || {
+        service::service_fn(create_response)
+    };
+
+    let server = Server::bind(&listen_address)
+        .serve(new_svc)
+        .map_err(|e| {
+            eprintln!("server error: {}", e);
+        });
+
+    println!("Listening on http://{}", listen_address);
+
+    // Run forever
+    hyper::rt::run(server);
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ed25519_dalek;
+    use std::convert::From;
+
+    #[test]
+    fn verify_request_smoke() {
+        // Get a legit request_hash signature, agent_id
+        let secret: [u8; 32] = [0_u8; 32];
+        let secret_key = ed25519_dalek::SecretKey::from_bytes( &secret ).unwrap();
+        let public_key = ed25519_dalek::PublicKey::from( &secret_key );
+        let secret_key_exp = ed25519_dalek::ExpandedSecretKey::from( &secret_key );
+
+        // Now lets sign some body payload
+        let body = json!({
+            "something": "interesting"
+        });
+        let body_json = serde_json::to_string( &body ).unwrap();
+
+        let signature = secret_key_exp.sign( body_json.as_bytes(), &public_key );
+        let mut signature_base64 = String::new();
+        base64::encode_config_buf(signature.to_bytes().as_ref(), base64::STANDARD, &mut signature_base64);
```

Comment:
```suggestion
        base64::encode_buf(signature.to_bytes().as_ref(), &mut signature_base64);
```

Reply:
this change === decision that we are using base64::STANDARD with padding

Reply:
Actually we are using base64::STANDARD_NO_PAD in verify_request(), so it should be `base64::encode_config_buf( ... , base64::STANDARD_NO_PAD, ... )`